libcst/tests/test_deep_replace.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from textwrap import dedent
+
+import libcst as cst
+from libcst.testing.utils import UnitTest
+
+
+class DeepCloneTest(UnitTest):
+    def test_deep_clone_simple(self) -> None:
+        old_code = """
+            pass
+        """
+        new_code = """
+            break
+        """
+
+        module = cst.parse_module(dedent(old_code))
+        pass_stmt = cst.ensure_type(module.body[0], cst.SimpleStatementLine).body[0]
+        new_module = cst.ensure_type(
+            module.deep_replace(pass_stmt, cst.Break()), cst.Module
+        )
+        self.assertEqual(new_module.code, dedent(new_code))
+
+    def test_deep_clone_identity(self) -> None:
+        old_code = """
+            pass
+        """
+        new_code = """
+            break
+        """
+
+        module = cst.parse_module(dedent(old_code))
+        new_module = module.deep_replace(
+            module,
+            cst.Module(
+                header=(cst.EmptyLine(),),
+                body=(cst.SimpleStatementLine(body=(cst.Break(),)),),
+            ),
+        )
+        self.assertEqual(new_module.code, dedent(new_code))
```

Comment:
Can we have a more advanced test case that there were multiple nodes of the same type and we successfully replace only one of them by identity?
E.g. replace the FunctionDef `c` as a new function.
```
def a():
    def b():
        def c():
            ...
        ...
    ...
```

Reply:
Good idea, I'll code that up real quick.